rconweb/api/views.py

```diff
@@ -82,30 +82,44 @@ def get_version(request):
 
 @csrf_exempt
 def public_info(request):
-    status = ctl.get_status()
+    gamestate = ctl.get_gamestate()
+    curr_players, max_players = tuple(map(int, ctl.get_slots().split("/")))
     try:
-        current_map = MapsHistory()[0]
+        current_map_start = MapsHistory(max_len=1)[0]["start"]
     except IndexError:
         logger.error("Can't get current map time, map_recorder is probably offline")
-        current_map = {"name": status["map"], "start": None, "end": None}
-    current_map = dict(
-        just_name=map_name(current_map["name"]),
-        human_name=LONG_HUMAN_MAP_NAMES.get(current_map["name"], current_map["name"]),
-        **current_map,
-    )
-    vote_status = get_votes_status(none_on_fail=True)
-    next_map = ctl.get_next_map()
+        current_map_start = None
+
+    def explode_map_info(game_map: str, start) -> dict:
+        return dict(
+            just_name = map_name(game_map),
+            human_name = LONG_HUMAN_MAP_NAMES.get(game_map, game_map),
+            name = game_map,
+            start = start
+        )
+
     return api_response(
-        result=dict(
-            current_map=current_map,
-            **status,
-            vote_status=vote_status,
-            next_map=next_map,
+        result = dict(
+            current_map = explode_map_info(gamestate["current_map"], current_map_start),
+            next_map = explode_map_info(ctl.get_next_map(), None),
+            player_count = curr_players,
+            max_player_count = max_players,
+            players = dict(
+                allied = gamestate["num_allied_players"],
+                axis = gamestate["num_axis_players"]
+            ),
+            score = dict(
+                allied = gamestate["allied_score"],
+                axis = gamestate["axis_score"]
+            ),
+            vote_status = get_votes_status(none_on_fail=True),
+            name = ctl.get_name(),
+            short_name = os.getenv("SERVER_SHORT_NAME", "HLL RCON"),
             public_stats_port=os.getenv('PUBLIC_STATS_PORT', "Not defined"),
             public_stats_port_https=os.getenv('PUBLIC_STATS_PORT_HTTPS', "Not defined")
         ),
-        failed=False,
-        command="public_info",
+        failed = False,
+        command = "public_info",
```

Comment:
The norm is no space for keyword arguments. 
I think we should run `black` on the whole codebase.

Reply:
Done, yes this sounds like a good idea!